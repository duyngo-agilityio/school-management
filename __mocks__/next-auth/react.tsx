import { createContext } from "react";

export const useSession = () => ({
  data: {
    user: {
      token: "access-token",
      email: "user@gmail.com",
      id: "1",
      name: "User",
    },
    expires: Date.now().toString(),
  },
  status: "authenticated",
  update: async () => null,
});

export type Session = {
  data: {
    user: {
      token: string;
      email: string;
      id: string;
      name: string;
    };
    expires: string;
  };
  status: "loading" | "authenticated" | "unauthenticated";
  update: (data?: Session) => Promise<Session | null>;
};

const INITIAL_SESSION: Session = {
  data: {
    user: {
      token: "access-token",
      email: "user@gmail.com",
      id: "1",
      name: "User",
    },
    expires: Date.now().toString(),
  },
  status: "authenticated",
  update: async () => null,
};

const SessionContext = createContext(INITIAL_SESSION);

export const SessionProvider = ({ children }: React.PropsWithChildren) => {
  <SessionContext.Provider value={INITIAL_SESSION}>
    {children}
  </SessionContext.Provider>;
};
