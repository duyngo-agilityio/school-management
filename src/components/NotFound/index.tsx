import Image from 'next/image';

// Components
import { Flex, Text } from '@chakra-ui/react';

interface NotFoundProps {
  title: string;
  src?: string;
  description?: string;
  width?: number;
  height?: number;
}

const NotFound = ({
  src = '',
  title,
  description,
  width = 340,
  height = 225,
}: NotFoundProps) => {
  return (
    <Flex flexDirection="column" justifyContent="center" alignItems="center">
      <Image src={src} alt="" width={width} height={height} />
      <Text variant="primary" fontWeight="semibold" size="xl" mt="31px">
        {title}
      </Text>
      <Text variant="primary" size="sm" mt="4px">
        {description}
      </Text>
    </Flex>
  );
};

export default NotFound;
