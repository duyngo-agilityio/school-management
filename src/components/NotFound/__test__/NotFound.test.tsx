import { render } from '@testing-library/react';

// Components
import NotFound from '..';

const NotFoundProps = {
  title: 'Page Not Found',
  src: '',
  description: 'Back home page',
  width: 340,
  height: 225,
};

describe('NotFound Component', () => {
  it('should match snapshot for NotFound', () => {
    const { container } = render(<NotFound {...NotFoundProps} />);

    expect(container).toMatchSnapshot();
  });
});
