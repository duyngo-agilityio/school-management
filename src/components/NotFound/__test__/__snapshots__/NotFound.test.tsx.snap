// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component should match snapshot for NotFound 1`] = `
<div>
  <div
    class="css-wkpw2c"
  >
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      height="225"
      src=""
      style="color: transparent;"
      width="340"
    />
    <p
      class="chakra-text css-1uy2b59"
    >
      Page Not Found
    </p>
    <p
      class="chakra-text css-1fhgjcy"
    >
      Back home page
    </p>
  </div>
</div>
`;
